source/docs/advanced/features/debugging.md

```diff
@@ -0,0 +1,98 @@
+
+## Debugging with Chrome DevTools
+
+### Client-side code
+
+Start your development server as usual by running `front-commerce start`, `npm run start`. Once the server starts, open `http://localhost:4000` (or your alternate URL) in Chrome. Next, open Chrome's Developer Tools (<kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>J</kbd> on Windows/Linux, <kbd>⌥</kbd>+<kbd>⌘</kbd>+<kbd>I</kbd> on macOS), then go to the **Sources** tab.
+
+Now, any time your client-side code reaches a [`debugger`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/debugger) statement, code execution will pause and that file will appear in the debug area. You can also press <kbd>Ctrl</kbd>+<kbd>P</kbd> on Windows/Linux or <kbd>⌘</kbd>+<kbd>P</kbd> on macOS to search for a file and set breakpoints manually.
+
+### Server-side code
+
+To debug server-side Front-Commerce code with Chrome DevTools, you need to pass the [`--inspect`](https://nodejs.org/api/cli.html#cli_inspect_host_port) flag to the underlying Node.js process:
+
+```bash
+cross-env NODE_ENV=development front-commerce start --inspect
+```
+
+If you're using `npm run start` then you should update the `start` script on your `package.json`:
+
+```json
+"start": "cross-env NODE_ENV=development front-commerce start --inspect"
+```
+
+Launching the Front-Commerce dev server with the `--inspect` flag will look something like this:
+
+```bash
+> front-commerce-skeleton@1.0.0-rc start
+> cross-env NODE_ENV=development front-commerce start "--inspect"
+
+Debugger listening on ws://127.0.0.1:9229/d00f8057-d9ce-4aeb-bcf2-805f7b681110
+For help, see: https://nodejs.org/en/docs/inspector
+   WEBPACK  Compiling client...
+   WEBPACK  Compiling server...
+```
+
+Once the server starts, open a new tab in Chrome and visit `chrome://inspect`, where you should see your Front-Commerce application inside the **Remote Target** section. Click **inspect** under your application to open a separate DevTools window, then go to the **Sources** tab.
+
+<!--
+TODO: Add development source maps with webpack?
+Debugging server-side code here works much like debugging client-side code with Chrome DevTools, except that when you search for files here with <kbd>Ctrl</kbd>+<kbd>P</kbd> or <kbd>⌘</kbd>+<kbd>P</kbd>, your source files will have paths starting with `webpack://{application-name}/./` (where `{application-name}` will be replaced with the name of your application according to your `package.json` file). -->
+
+## More information
+
+To learn more about how to use a JavaScript debugger, take a look at the following documentation:
+
+- [Node.js debugging in VS Code: Breakpoints](https://code.visualstudio.com/docs/nodejs/nodejs-debugging#_breakpoints)
+- [Chrome DevTools: Debug JavaScript](https://developers.google.com/web/tools/chrome-devtools/javascript)
```

Comment:
Firefox has not been included because it does not support node.js debugging which was the main point of this documentation.

https://stackoverflow.com/a/56759070

Reply:
yeah I know but it works perfectly for the Frontend and unless we want a webkit only world, I think it should be mentioned here.

Reply:
Add reference to firefox as well